Add XML support and certificate authentication for Kafka connector

ga-hegsvold opened the pull request.

ga-hegsvold committed:
Add XML support

ga-hegsvold committed:
Update requirements for XML support

ga-hegsvold committed:
Merge pull request #1 from ga-hegsvold/ga-hegsvold-patch-1

Add XML support

ga-hegsvold committed:
Turn off debug mode

BaardBouvet commented:
Another way to solve this would be to base64 encode the content and add the content-type to the entity in JSON (add support for that in this microservice), and then have another external transform that decodes base64 encoded xml to json. This would be a more flexible and reusable pattern.

ga-hegsvold commented:
> Another way to solve this would be to base64 encode the content and add the content-type to the entity in JSON (add support for that in this microservice), and then have another external transform that decodes base64 encoded xml to json. This would be a more flexible and reusable pattern.

@BaardBouvet 
Could you elaborate a bit?
You mean to take the XML input, base64 encoded it and put it and the content-type into two new JSON properties?
Then write another MS to decode the XML based on the content-type and transform it to JSON?

ga-hegsvold committed:
Update service.py

ga-hegsvold committed:
Update service.py

ga-hegsvold commented:
@BaardBouvet Did some changes based on your feedback. Can you re-review?

ga-hegsvold committed:
Remove redundant requirements

ga-hegsvold committed:
Remove deprecated import

ga-hegsvold commented:
@BaardBouvet review please.

ga-hegsvold committed:
Certificate authentication support

- New module certificate_handler.py to write certificates to file
- Can now handle both credentials and certificate authentication
- Switched to 'production' mode with cherrypy

ga-hegsvold committed:
Merge pull request #2 from ga-hegsvold/cert-handler

Certificate authentication support

ga-hegsvold committed:
Fix typos

ga-hegsvold committed:
Merge pull request #3 from ga-hegsvold/cert-handler

Fix typos

ga-hegsvold committed:
Update comments & README

ga-hegsvold committed:
Merge pull request #4 from ga-hegsvold/cert-handler

Update comments & README

ga-hegsvold committed:
Update old example configs

ga-hegsvold committed:
Merge pull request #5 from ga-hegsvold/cert-handler

Update old example configs

ga-hegsvold commented:
@BaardBouvet Added certificate authentication handling and updated documentation. Please review changes.
Both credential- and certificate authentication have been tested at Statnett.

ga-hegsvold commented:
@BaardBouvet Can you approve this one?
